### PR TITLE
Changed almanac functions to take const pointers.

### PR DESCRIPTION
**Pull request by ungureanuvladvictor**
Signed-off-by: Vlad Ungureanu <vvu@vdev.ro>

### CODE REVIEW

File: include/libswiftnav/almanac.h [hunk 1]
```diff
@@ -39,9 +39,9 @@ typedef struct {
 
 /** \} */
 
-void calc_sat_state_almanac(almanac_t* alm, double t, s16 week,
+void calc_sat_state_almanac(const almanac_t* alm, double t, s16 week,
                             double pos[3], double vel[3]);
-void calc_sat_az_el_almanac(almanac_t* alm, double t, s16 week,
+void calc_sat_az_el_almanac(const almanac_t* alm, double t, s16 week,
                             double ref[3], double* az, double* el);
 double calc_sat_doppler_almanac(almanac_t* alm, double t, s16 week,
```
Comment (gsmcmullin) [inline include/libswiftnav/almanac.h]: Why not make this alm const too?

**Comment by gsmcmullin** [comment]
Looks good.  There are probably a lot of other missing const's around too.

**Comment by henryhallam** [comment]
Good find. While you're at it, why not make the signatures of the functions fully const-correct? i.e. add `const` to `ref`.

**Comment by mookerji** [comment]
+1 to this PR and Henry's suggestion as well.

**Comment by mookerji** [comment]
While you're at it, fixing const correctness throughout the entire project wouldn't be terrible. Henry's already started some of this here: https://github.com/swift-nav/libswiftnav/pull/150/files

**Comment by ungureanuvladvictor** [comment]
I will continue later on the const work. This PR will remain like this, I will do another one to refactor the remaining const things.

**Pull request merged**